internal/primitive/transform/function/strings_functions.go

```diff
@@ -120,6 +120,38 @@ var CapitalizeSentence = function{
 	},
 }
 
+var ReplaceBetweenDelimitersFunction = function{
+	name:		"REPLACE_BETWEEN_DELIMITERS",
+	fixedArgs:	[]common.Type{common.String, common.String, common.String, common.String},
+	fn:	func(args []interface{}) (interface{}, error) {
+		value, _ := args[0].(string)
+		startPattern := args[1].(string)
+		endPattern := args[2].(string)
+		newValue := args[3].(string)
+
+		switch {
+			case startPattern != endPattern && strings.Contains(value, startPattern) && strings.Contains(value, endPattern):
+				if strings.Index(value, startPattern) > strings.Index(value, endPattern) {
+					return nil, fmt.Errorf("the end pattern is before the start pattern in the input string")
+				}
+				firstSplit := strings.Split(value, startPattern)
+				secondSplit := strings.Split(firstSplit[1], endPattern)
+				secondSplit[0] = startPattern + newValue + endPattern
+				return firstSplit[0] + secondSplit[0] + secondSplit[1], nil
+			case startPattern == endPattern && strings.Contains(value, startPattern) && strings.Count(value, startPattern) == 2:
+				firstSplit := strings.Split(value, startPattern)
+				firstSplit[1] = startPattern + newValue + endPattern
+				return firstSplit[0] + firstSplit[1] + firstSplit[2], nil
+			case strings.Contains(value, startPattern) && strings.Contains(value, endPattern):
```

Comment:
the line 145 is same lint 147

Reply:
my apologies for that, I was supposed to make one case check for startPattern present and endPattern false and the other way round for the other case.